packages/server/tests/erc20Contract/ERC20Contract.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+
+contract ERC20Contract {
+
+    function name(address token) public view {
+        IERC20Metadata(token).name();
+    }
+
+    function symbol(address token) public view {
+        IERC20Metadata(token).symbol();
+    }
+
+    function decimals(address token) public view {
+        IERC20Metadata(token).decimals();
+    }
+
+    function totalSupply(address token) public view {
+        IERC20(token).totalSupply();
+    }
+
+    function balanceOf(address token, address account) public view {
+        IERC20(token).balanceOf(account);
+    }
+
+    function transfer(address token, address recipient, uint256 amount) public {
+        IERC20(token).transfer(recipient, amount);
+    }
+
+    function delegateTransfer(address token, address recipient, uint256 amount) public {
+        (bool success, bytes memory result) = address(IERC20(token)).delegatecall(abi.encodeWithSignature("transfer(address,uint256)", recipient, amount));
+    }
+
+    //Not supported operations - should return a failure
+
+    function transferFrom(address token, address sender, address recipient, uint256 amount) public {
```

Comment:
transferFrom / approval fails because you have `approved` the `msg.sender` to spend tokens, but after that when you call `transferFrom` you are spending them on the `erc20contract` behalf